axelor-stock/src/main/java/com/axelor/apps/stock/web/StockMoveController.java

```diff
@@ -662,4 +662,26 @@ public void changeLinesToStockLocation(ActionRequest request, ActionResponse res
     Beans.get(StockMoveService.class).changeLinesToStockLocation(stockMove, toStockLocation);
     response.setValue("stockMoveLineList", stockMove.getStockMoveLineList());
   }
+
+  public void generateNewStockMove(ActionRequest request, ActionResponse response) {
+
+    try {
+      StockMove stockMove = request.getContext().asType(StockMove.class);
+      stockMove = Beans.get(StockMoveRepository.class).find(stockMove.getId());
+      Optional<StockMove> newStockMove =
+          Beans.get(StockMoveService.class).generateNewStockMove(stockMove);
+      if (newStockMove.isPresent()) {
+        response.setView(
+            ActionView.define(I18n.get("Stock move"))
+                .model(StockMove.class.getName())
+                .add("grid", "stock-move-out-grid")
```

Comment:
Please check the type of the generated stock move to set the appropriate grid view